runtime/common/BaseRestRemoteClient.h

```diff
@@ -276,6 +287,15 @@ class BaseRemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
               const std::string &backendSimName, const std::string &kernelName,
               void (*kernelFunc)(void *), void *kernelArgs,
               std::uint64_t argsSize, std::string *optionalErrorMsg) override {
+    if (isDisallowed(io_context.name)) {
```

Comment:
It might be a good idea to add a test to `python/tests/remote/test_remote_platform.py` to test that the error message is indeed fired. If it is easier to change the std::cout to a std::runtime_exception for the purposes of pytest, that would probably ok.